### ISSUE TITLE
Handle decoding of XML/HTML entities in the SAX parser

### ISSUE THREAD

**Issue by YorickPeterse** [YorickPeterse/oga#92]
Because the SAX parser doesn't rely on `XML::Text` and `XML::Attribute` this means that currently no entities are decoded. Decoding XML entities is fairly simple. Decoding HTML entities on the other hand will be much more difficult as the SAX parser doesn't keep track of any state that can be used if we're

1. In an HTML document
2. Not in a literal element (script/style tags)

The lazy way out is to let the user handle this, but this isn't going to make the SAX parsing API much more pleasant to use.

**Comment by YorickPeterse** [comment]
@krasnoukhov commit da62fcd75d0889e4539e7390777a906a914a78c0 adds decoding of XML/HTML entities in the SAX parser. Could you run sax-machine against this commit to see if this fixes the entity issues you were experiencing (it should, but it's worth double checking)?

**Comment by krasnoukhov** [comment]
Thanks for the notice. In fact, everything works great, please proceed with new release!

**Comment by YorickPeterse** [comment]
This has been released in version 0.3.4.

**Comment by krasnoukhov** [comment]
Thanks! By the way, here is a small thing that I've noticed. Decoding an attribute with a `&#038` in it throws the `ArgumentError`:

```
lib/oga/xml/entities.rb:75:in `Integer': invalid value for Integer(): "038" (ArgumentError)
```

I know that it's invalid use, but it might make sense to address this somehow.

**Comment by YorickPeterse** [comment]
@krasnoukhov Please report that as a separate issue as it's not directly related to this one.

**Comment by YorickPeterse** [comment]
@krasnoukhov Nevermind, this has been fixed in 853d804f3468c9f54c222568a7faedf736f8dc1a.

**Comment by krasnoukhov** [comment]
Wow, that was fast! Thank you
